cli/lib/build.js

```diff
@@ -52,6 +52,54 @@ const checkBranch = async () => {
   }
 };
 
+const checkCodeUpToDate = async () => {
+  logInfo('update code');
+
+  let answer = await inquirer.prompt([
+    {
+      type: 'confirm',
+      name: 'updateCode',
+      message: 'Whether to auto update the code? Or you can update manually and then select "N" to continue.',
+      default: false,
+    },
+  ]);
+
+  if (answer.updateCode) {
+    logInfo(`performing "git pull upstream develop" inside "${rootDir}"`);
+    await execSync(`git pull upstream develop`, { cwd: rootDir });
+
+    let enterpriseDir = path.resolve(rootDir, '../erda-ui-enterprise');
+
+    if (!fs.existsSync(enterpriseDir)) {
+      answer = await inquirer.prompt([
+        {
+          type: 'directory',
+          name: 'targetPath',
+          message: 'Select erda-ui-enterprise directory',
+          basePath: path.resolve(rootDir, '..')
+        }
+      ]);
+
+      enterpriseDir = answer.targetPath;
+    }
+    logInfo(`performing "git pull upstream master" inside "${enterpriseDir}"`);
+    await execSync(`git pull upstream master`, { cwd: enterpriseDir });
+
+    answer = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'solvedConflicts',
+        message: 'Have all merge conflicts been solved?',
+        default: false,
```

Comment:
You can use a package such as `git-status` to check if there is any conflict or uncommitted changes.